actions/application_event.py

```diff
@@ -25,6 +30,42 @@ def create_reservations_for_event(self) -> None:
         if status != ApplicationEventStatusChoice.APPROVED:
             raise ValidationError(_(f"Cannot create reservations for event based on its status: '{status.value}'"))
 
-        schedule: "ApplicationEventSchedule"
+        schedule: ApplicationEventSchedule
         for schedule in self.application_event.application_event_schedules.all().accepted():
             schedule.actions.create_reservation_for_schedule()
+
+    def application_events_affecting_allocations(self) -> ApplicationEventQuerySet:
+        from applications.models import ApplicationEvent
+        from reservation_units.models import ReservationUnit
+
+        return (
+            ApplicationEvent.objects.distinct()
+            .alias(
+                _affecting_ids=models.Subquery(
+                    queryset=(
+                        ReservationUnit.objects.alias(
+                            # All reservation units that are possible for this application event
+                            _reservation_unit_ids=models.Subquery(
+                                self.application_event.event_reservation_units.values("reservation_unit__id"),
+                            ),
+                        )
+                        .filter(id__in=models.F("_reservation_unit_ids"))
+                        .with_reservation_unit_ids_affecting_reservations()
+                        .annotate(_found_ids=ArrayUnnest("reservation_units_affecting_reservations"))
+                        .values("_found_ids")
+                    )
+                )
+            )
+            .filter(
+                # Don't include this event.
+                ~models.Q(id=self.application_event.id),
+                # Application event has schedules.
+                application_event_schedules__isnull=False,
+                # Schedule has been allocated to any reservation unit affecting this event's allocations.
+                application_event_schedules__allocated_reservation_unit__isnull=False,
+                application_event_schedules__allocated_reservation_unit__id__in=models.F("_affecting_ids"),
+                # Allocation period overlaps with this event's period.
+                application_event_schedules__application_event__begin__lt=self.application_event.end,
+                application_event_schedules__application_event__end__gt=self.application_event.begin,
```

Comment:
Should these be `__lte` and `__gte` instead?

Reply:
I guess they should since it's dates and not datetimes. Normal logic for overlapping datetimes should be `lt` and `gt`, since we don't care if the datetimes are right next to each other, but for dates it does matter, since the whole day is considered.

Reply:
Done